File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlValue.cs
using YamlDotNet.Core.Tokens;

namespace T1.SqlSharp.Expressions;

public class SqlValue : ISqlExpression
{
    public SqlType SqlType { get; set; } = SqlType.String;
    public string Value { get; set; } = string.Empty;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_SqlValue(this);
    }

    public string ToSql()
    {
        return $"{Value}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlValue.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlValue implements ISqlExpression {
    SqlType: SqlType = SqlType.String;
    Value: string = '';
    Span: TextSpan = new TextSpan();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_SqlValue(this);
    }

    ToSql(): string {
        return `${this.Value}`;
    }
} 